src/Network/Minio/Errors.hs

```diff
@@ -41,11 +41,22 @@ instance Exception MErrV
 
 -- | Errors returned by S3 compatible service
 data ServiceErr = BucketAlreadyExists
+                | BucketAlreadyOwnedByYou
                 | NoSuchBucket
+                | NoSuchKey
+                | ServiceErr Text Text
   deriving (Show, Eq)
 
 instance Exception ServiceErr
 
+toServiceErr :: Text -> Text -> ServiceErr
+toServiceErr "NoSuchKey" _               = NoSuchKey
+toServiceErr "NoSuchBucket" _            = NoSuchBucket
+toServiceErr "BucketAlreadyOwnedByYou" _ = BucketAlreadyOwnedByYou
+toServiceErr "BucketAlreadyExists" _     = BucketAlreadyExists
+toServiceErr code message                = ServiceErr code message
```

Comment:
Add some live server tests - may be one for each case. Not sure if Minio ever throws BucketAlreadyExists rather than BucketAlreadyOwnedByYou.